### PR TITLE
Add system test reproducing merge edge case bug with source-only replicas

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review
@toregge @baldersheim FYI

This reproduces the edge case bug causing potential data loss when
all of the following conditions hold true:

* The amount of document data to be merged exceeds the configured per-diff chunk size
* There is more than 1 source-only node involved in the merge chain
* The document sets contained on the source-only nodes are at least partially disjoint

Test sets up 3 nodes with entirely disjoint data sets, artificially
disabling merges while disjoint feeding takes place. 2 out of 3 nodes
are marked as Retired, enforcing a multi-node source-only merge for
data to be transferred to the remaining node.

Increasing the chunk size limit beyond the bucket size causes the bug
to _not_ trigger.

### CODE REVIEW

File: tests/search/inconsistent_replicas/disjoint_source_only_documents.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+require 'search_test'
+require 'doc_generator'
+
+class DisjointSourceOnlyDocuments < SearchTest
+
+  def make_app(disable_merges:)
+    SearchApp.new.sd(SEARCH_DATA + 'music.sd').
```
Comment (geirst) [inline tests/search/inconsistent_replicas/disjoint_source_only_documents.rb:8]: Consider creating a `test.sd` instead with just the title field needed.

**Review by geirst — APPROVED** [review]
👍

File: tests/search/inconsistent_replicas/disjoint_source_only_documents.rb (hunk 1 above)
Comment (vekterli) [reply tests/search/inconsistent_replicas/disjoint_source_only_documents.rb]: Fixed

File: tests/search/inconsistent_replicas/test.sd [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+schema test {
+  
+  document test {
+
+    field title type string {
+      indexing: index | summary
```
Comment (geirst) [inline tests/search/inconsistent_replicas/test.sd:8]: Nitpick: `summary` is sufficient here.
